auth/opa/rpcauth/rpcauth.go

```diff
@@ -80,32 +80,38 @@ func NewWithPolicy(ctx context.Context, policy string, authzHooks ...RPCAuthzHoo
 // the success or failure of policy.
 func (g *Authorizer) Eval(ctx context.Context, input *RPCAuthInput) error {
 	logger := logr.FromContextOrDiscard(ctx)
+	logger = logger.WithValues("input", input)
 	if input != nil {
 		if logger.V(2).Enabled() {
-			logger.V(2).Info("evaluating authz policy", "input", input)
+			logger.V(2).Info("evaluating authz policy")
 		}
 	}
 	if input == nil {
-		return status.Error(codes.InvalidArgument, "policy input cannot be nil")
+		err := status.Error(codes.InvalidArgument, "policy input cannot be nil")
+		logger.V(1).Error(err, "failed to evaluate authz policy")
+		return err
 	}
 	for _, hook := range g.hooks {
 		if err := hook.Hook(ctx, input); err != nil {
+			logger.V(1).Error(err, "authz hook error")
 			if _, ok := status.FromError(err); ok {
 				// error is already an appropriate status.Status
 				return err
 			}
 			return status.Errorf(codes.Internal, "authz hook error: %v", err)
 		}
 	}
-	if logger.V(1).Enabled() {
-		logger.V(1).Info("evaluating authz policy post hooks", "input", input)
+	if logger.V(2).Enabled() {
```

Comment:
Can you remove the if statement?

From looking at some history, it's leftover from when we used to do more work inside the statement prior to https://github.com/Snowflake-Labs/sansshell/pull/151

Reply:
Done.